In [2]:
import pandas as pd 
import numpy as np

In [3]:
annotation = pd.read_csv("data/df_expanded.csv")
annotation

first_num              perovskite_composition electron_transport_layer  \
0            0        Cs 0.05 FA 0.85 MA 0.1 PbI 3                      C60   
1            1                                 NaN                 TinOxide   
2            2         dibutylammonium lead iodide              tin dioxide   
3            2         dibutylammonium lead iodide              tin dioxide   
4            2         dibutylammonium lead iodide              tin dioxide   
..         ...                                 ...                      ...   
229        147           formamidinium lead iodide                      NaN   
230        147           formamidinium lead iodide                      NaN   
231        148  Cs0.05FA0.85MA0.10Pb(I0.97Br0.03)3                      NaN   
232        148  Cs0.05FA0.85MA0.10Pb(I0.97Br0.03)3                      NaN   
233        149              FA0.7MA0.3Pb0.5Sn0.5I3     buckminsterfullerene   

                                  hole_transport_layer structure_pin_nip  \
0                                   2PACz and Me-4PACz               PIN   
1    poly[bis(4-phenyl) (2,4,6-trimethylphenyl)amin...               PIN   
2                                         Spiro-OMeTAD               NIP   
3                                         Spiro-OMeTAD               NIP   
4                                         Spiro-OMeTAD               NIP   
..                                                 ...               ...   
229                                       Spiro-OMeTAD               NIP   
230                                       Spiro-OMeTAD               NIP   
231                                       Spiro-OMeTAD               NIP   
232                                       Spiro-OMeTAD               NIP   
233                                                NaN               NaN   

         test stability_type  \
0      test_1          ISOSL   
1      test_1          ISOST   
2      test_1          ISOSL   
3    test_1_2          ISOSD   
4      test_2         ISOSLT   
..        ...            ...   
229    test_1          ISOSD   
230    test_2         ISOSLT   
231  test_1_2         ISOSLT   
232    test_1            NaN   
233    test_1         ISOSLT   

                                  passivating_molecule  humidity  temperature  \
0                   4-chlorobenzenesulfonate (4Cl-BZS)       NaN         65.0   
1                                    phenethylammonium       NaN         85.0   
2                                           butylamine      85.0         25.0   
3                                                  NaN      85.0         85.0   
4                                                  NaN       NaN         25.0   
..                                                 ...       ...          ...   
229  carbazole-triphenylamine and phenylammonium io...      85.0         85.0   
230                                                NaN       NaN          NaN   
231                         phenylethylammonium iodide     507.0          NaN   
232                 4-tert-butyl-benzylammonium iodide       NaN          NaN   
233                   4-trifluoromethyl-phenylammonium     305.0         35.0   

       time  control_pce  treated_pce  control_voc  treated_voc  \
0    1200.0         24.0        26.90          NaN         1.18   
1     500.0          NaN        19.10          NaN         1.16   
2    1620.0          NaN        24.30          NaN         1.18   
3    1056.0          NaN        21.30          NaN          NaN   
4    1620.0          NaN        24.00          NaN          NaN   
..      ...          ...          ...          ...          ...   
229  1000.0         22.3          NaN          NaN         1.11   
230  1100.0          NaN        24.70          NaN          NaN   
231   500.0          NaN         0.84          NaN          NaN   
232     NaN         21.2        22.70         1.09         1.12   
233   600.0          NaN        26.40          N

In [4]:
data_inspect = annotation['efficiency_tret'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)


0.9
0.95
0.8
1.0
0.92
0.96
0.91
0.98
0.85
0.94
0.97
0.86
0.99
0.93
0.75
0.74
0.88
0.82
0.87
0.6
0.53
0.67
98.9
0.89
0.72
0.45
0.65
95.0
0.47
96.7
0.83
0.81
0.35
0.73
0.84
1.02
0.71


## Noting any weird value in each column

# efficiency_tret

In [52]:
data_inspect = annotation['efficiency_tret'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

0.9
0.95
0.8
1.0
0.92
0.96
0.91
0.98
0.85
0.94
0.97
0.86
0.99
0.93
0.75
0.74
0.88
0.82
0.87
0.6
0.53
0.67
98.9
0.89
0.72
0.45
0.65
95.0
0.47
96.7
0.83
0.81
0.35
0.73
0.84
1.02
0.71


efficiency_tret
- 96.7
- 98.9
- 1.02

In [ ]:
efficiency_tret = annotation[["first_num","efficiency_tret"]]
efficiency_tret = efficiency_tret[efficiency_tret['efficiency_tret'] == 0.89]
efficiency_tret

# efficiency_cont

In [41]:
data_inspect = annotation['efficiency_cont'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

0.6
0.5
0.7
0.4
0.8
0.0
0.58
0.65
0.27
0.68
0.63
0.84
0.75
0.3
0.2
0.38
0.71
0.76
0.85
0.57
0.77
0.87
0.16
0.55
0.73
0.9
0.83
0.56
0.82
0.78
0.98
0.1
0.89
0.96
0.92
0.29
0.51
0.47
0.74
0.95
0.94
0.09
0.25
0.67
0.59
0.49
0.32
0.93
0.44
0.66


efficiency_cont
- 0.1
- 0.96
- 0.89
- 0.82


In [45]:
efficiency_cont = annotation[["first_num","efficiency_cont"]]
efficiency_cont = efficiency_cont[efficiency_cont['efficiency_cont'] == 0.89]
efficiency_cont

first_num  efficiency_cont
228        146             0.89

# temperature

In [29]:
data_inspect = annotation['temperature'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

85.0
25.0
65.0
60.0
40.0
50.0
55.0
80.0
35.0
150.0
75.0
250.0
120.0
254.0
70.0
30.0
508.0
450.0
45.0
23.0
28.0
20.0
21.0
100.0
27.0
52.0
110.0
2000.0


temperature
- 2000.0
- 508.0
- 150.0
- 450.0
- 250.0
- 120.0
- 254.0
- 110
- 100

In [39]:
temperature = annotation[["first_num","temperature"]]
temperature = temperature[temperature['temperature'] == 100]
temperature

first_num  temperature
73         63        100.0

# humidity

In [5]:
data_inspect = annotation['humidity'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

85.0
50.0
25.0
65.0
20.0
30.0
60.0
40.0
55.0
152.0
203.0
75.0
304.0
255.0
70.0
507.0
506.0
37.0
0.25
0.85
558.0
601.0
5.0
80.0
405.0
305.0


humidity
- 152.0
- 203.0
- 304.0
- 255.0
- 507.0
- 506.0
- 0.25
- 558.0
- 601.0
- 0.85
- 5.0
- 405.0
- 305.0

In [26]:
humidity = annotation[["first_num","humidity"]]
humidity = humidity[humidity['humidity'] == 305]
humidity

first_num  humidity
233        149     305.0

# Passivating Molecule

In [96]:
data_inspect = annotation['passivating_molecule'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

methylammonium lead iodide
ISOSLT
fluorophenylethylammonium iodide
butylammonium
oleylammonium iodide
poly(methyl methacrylate)
NIP
phenylethylammonium iodide
(EDBE)PbI 4
4-fluoroaniline
europium ion pair
MAPbI 3
FAPbI3
carbazole-triphenylamine and phenylammonium iodide units
formamidinium (FA)
HTAB
FA0.5MA0.5PbI3
quanternary ammonium halides
4-tert-butyl-benzylammonium iodide
Methylammonium Lead Bromide
(phenethylamino)methaniminium iodide
4-fluorophenylethylammonium iodide
CF3 -PEAI
Formamidinium Lead Iodide
ISOST
glycine hydrochloride
chlorine-rich mixed-halide perovskite interlayer
lead iodide
Piperazinium Diiodide
piperazinium iodide
Titanium dioxide
Butylammonium bromide
ferrocenyl-bis-thiophene-2-carboxylate
ortho-carborane
lead oxalate
zinc-based halogenometallate
sodium thioglycolate
methylammonium iodide
3-diammonium iodide,propane-1
formate
4-chlorobenzenesulfonate (4Cl-BZS)
PCBM
cyclohexylmethylammonium iodide dissolved in
butylamine
iso-butylamine iodide
ethylammonium pyre


passivating_molecule fix

- methylammonium lead iodide
- ISOSLT
- NIP
- MAPbI 3
- FAPbI3
- formamidinium (FA)
- FA0.5MA0.5PbI3
- Methylammonium Lead Bromide
- Formamidinium Lead Iodide
- ISOST
- cyclohexylmethylammonium iodide dissolved in
- intermediate negative ΔE vac
- NOTE:This is how to prepare the PSC,relevent...?
- SnO2
- 22.38

methylammonium iodide
phenylethylammonium iodide
fluorophenylethylammonium iodide
butylammonium
oleylammonium iodide
poly(methyl methacrylate)
(EDBE)PbI 4
4-fluoroaniline
europium ion pair
carbazole-triphenylamine and phenylammonium iodide units
HTAB
quanternary ammonium halides
4-tert-butyl-benzylammonium iodide
(phenethylamino)methaniminium iodide
4-fluorophenylethylammonium iodide
CF3 -PEAI
glycine hydrochloride
chlorine-rich mixed-halide perovskite interlayer
lead iodide
Piperazinium Diiodide
piperazinium iodide
Titanium dioxide
Butylammonium bromide
ferrocenyl-bis-thiophene-2-carboxylate
ortho-carborane
lead oxalate
zinc-based halogenometallate
sodium thioglycolate
3-diammonium iodide,propane-1
formate
4-chlorobenzenesulfonate (4Cl-BZS)
PCBM
butylamine
iso-butylamine iodide
ethylammonium pyrene
EDBE
OABF4
benzotriazole
butylammonium bromide
2-thiopheneethylammonium chloride
chlorophenylethylammonium iodide
CF3-phenethylammonium
2-thiophenemethylammonium iodide
3,4,5-trifluoroanilinium
phenylethylammonium lead iodide
phenethylammonium iodide
trifluoromethylphenylethylammonium
3-(aminomethyl)pyridine
tri-octyl phosphine oxide
1,3-propylene diammonium
2-thiophenemethylammonium bromide
azetidinium lead iodide
Lithium Fluoride
PS Plastic Foam 1%
n-butylamine iodide
DMePDAI 2
methylammonium
phenethylammonium
4-trifluoromethyl-phenylammonium


In [50]:
passivatin = annotation[["first_num","passivating_molecule"]]
passivatin = passivatin[passivatin['passivating_molecule'] == "PS Plastic Foam 1%"]
passivatin

first_num passivating_molecule
53         48   PS Plastic Foam 1%

"PS Plastic Foam 1%" -- Paper 48, No perovskite composition was scraped, skip. 

# PIN NIP structure

In [92]:
data_inspect = annotation['structure_pin_nip'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

NIP
PIN
NiP
secondary perovskite growth
poly(triarylamine) (PTAA)/ indium tin oxide (ITO)/glass substrates
porous insulator contact


structure_pin_nip
- secondary perovskite growth
- poly(triarylamine) (PTAA)/ indium tin oxide (ITO)/glass substrates
- porous insulator contact

In [95]:
structure = annotation[["first_num","structure_pin_nip"]]
structure = structure[structure['structure_pin_nip'] == "porous insulator contact"]
structure

first_num         structure_pin_nip
198        131  porous insulator contact

# HTL

In [89]:
data_inspect = annotation['hole_transport_layer'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

Spiro-OMeTAD
PTAA
poly(triarylamine)
11-dimethoxy-7H-dibenzo[c,4-(3,g]carbazol-7-yl)butyl)phosphonic acid
polybisphenyl(trimethylphenyl)amine
6-dimethoxy-9H-carbazol-9-yl)ethyl]phosphonic acid,[2-(3
polymeric poly(4-vinylpyridine)
bis(trifluoromethane)sulfonimide lithium salt/4-tert-butylpyridine (LiTFSI/tBP)doped poly(triarylamine)
ion-modulated radical doping Spiro-OMeTAD
MeO-2PACZ
spiro-OMeTAD/chlorobenzene
DDT-treated spiro-OMeTAD
(2,7-Dimethoxy-9H-carbazol-9-yl)methyl phosphonic acid
poly(triaryl)amine
PEDOT:PSS
2,2'-(9,6-diyl)bis(4-(1H-pyrazol-1-yl)-phenyl)amine,9-dimethyl-9H-carbazol-3
2PACz
spiro-MeOTAD
P3CT-N
Me-4PACz
2PACz and Me-4PACz
poly(triaryl amine)
Spiro[OMeTAD-3-fluorophenyl]
poly[bis(4-phenyl) (2,4,6-trimethylphenyl)amine] (PTAA)
TaTm
copper(I) thiocyanate
Poly(3-hexylthiophene)


hole_transport_layer
- PTAA
- MeO-2PACZ
- Me-4PACz
- 2PACz
- PEDOT:PSS

In [91]:
HTL = annotation[["first_num","hole_transport_layer"]]
HTL = HTL[HTL['hole_transport_layer'] == "poly(triarylamine)"]
HTL

first_num hole_transport_layer
40          36   poly(triarylamine)
65          57   poly(triarylamine)
164        114   poly(triarylamine)
165        114   poly(triarylamine)
191        127   poly(triarylamine)
192        128   poly(triarylamine)
193        128   poly(triarylamine)

# ETL

In [80]:
data_inspect = annotation['electron_transport_layer'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

buckminsterfullerene
TinOxide
tin dioxide
Titanium dioxide
TitaniumDioxide
PCBM
C60
6] -phenyl-C61-butyric acid methyl ester,[6
Lanthanum oxide / Tin Oxide Bilayer
ISOSLT
indium tin oxide
Tin Oxide
titanium oxynitride
polyacrylic acid-stabilized tin(IV) oxide quantum dots
C 60
boron chloride subphthalocyanine
bathocuproine
ligand allylammonium
SnO 2
SnO2
Cl-bonded SnO 2


electron_transport_layer

- TinOxide
- tin dioxide
- Titanium dioxide
- TitaniumDioxide
- PCBM
- C60
- Lanthanum oxide / Tin Oxide Bilayer
- indium tin oxide
- Tin Oxide
- titanium oxynitride
- polyacrylic acid-stabilized tin(IV) oxide quantum dots
- C 60
- boron chloride subphthalocyanine
- bathocuproine
- ligand allylammonium
- SnO 2
- SnO2
- Cl-bonded SnO 2




- buckminsterfullerene --> C60 
- ISOSLT
- 6] -phenyl-C61-butyric acid methyl ester,[6 ---> [6, 6] -phenyl-C61-butyric acid methyl ester
- polyacrylic acid-stabilized tin(IV) oxide quantum dots

In [88]:
ETL = annotation[["first_num","electron_transport_layer"]]
ETL = ETL[ETL['electron_transport_layer'] == "6] -phenyl-C61-butyric acid methyl ester,[6"]
ETL

first_num                     electron_transport_layer
182        124  6] -phenyl-C61-butyric acid methyl ester,[6
183        124  6] -phenyl-C61-butyric acid methyl ester,[6
184        124  6] -phenyl-C61-butyric acid methyl ester,[6
185        124  6] -phenyl-C61-butyric acid methyl ester,[6

efficiency_tret
- 96.7
- 98.9
- 1.02

efficiency_cont
- 0.1

temperature
- 2000.0
- 508.0
- 150.0
- 450.0
- 250.0
- 120.0
- 254.0




humidity
- 152.0
- 203.0
- 304.0
- 255.0
- 507.0
- 506.0
- 0.25
- 558.0
- 601.0
- 0.85
- 5.0
- 405.0
- 305.0


passivating_molecule
- cyclohexylmethylammonium iodide dissolved in
- CF3 -PEAI
- NIP
- ISOSLT
- ISOST
- FAPbI3
- PCBM
- HTAB
- EDBE
- OABF4
- intermediate negative ΔE vac
- NOTE:This is how to prepare the PSC,relevent...?

structure_pin_nip
- secondary perovskite growth
- poly(triarylamine) (PTAA)/ indium tin oxide (ITO)/glass substrates
- porous insulator contact

hole_transport_layer
- PTAA
- MeO-2PACZ
- Me-4PACz
- 2PACz
- PEDOT:PSS

electron_transport_layer
- buckminsterfullerene
- ISOSLT
- polyacrylic acid-stabilized tin(IV) oxide quantum dots

# Perovskite_composition

In [64]:
data_inspect = annotation['perovskite_composition'].value_counts()

for index, value in data_inspect.iteritems():
    print(index)

formamidinium lead iodide
(FAPbI3)0.95(MAPbBr3)0.05
FA0.83Cs0.17PbI2.7Br0.3
FA0.85MA0.1Cs0.05PbI2.9Br0.1
Formamidinium lead triiodide
CH3NH3PbI3
Cs0.1FA0.9PbI3
FA0.98Cs0.02PbI3
FA0.98MA0.02Pbl3
Cs 0.05(FA0.98MA0.02)0.95Pb(I0.98Br0.02)3
dibutylammonium lead iodide
formamidine formate
CH 3 NH 3 PbI 3
576
methylammonium lead iodide
FAPbI 3 with 0.03 mol fraction of both MDA and Cs cations
Methylammonium Lead Bromide
ISOSD3
Cs0.05(FAPbI3)0.85(MAPbBr3)0.15
CaFAMA-F
Cs0.15FA0.85PbI2.55Br0.45
CsMAFA
Cs 0.05(FA0.95MA0.05)0.95Pb(I0.95Br0.05)3
(FAPbI 3 ) 0.95 (MAPbBr 3 ) 0.05
FA0.83MA0.17Pb-(I0.83Br0.17)3
Cs 0.05 (FA0.92MA0.08)0.95Pb(I0.92Br0.08)3
Cs 0.12 FA 0.8 MA 0.08 PbI 1.8 Br 1.
Cs0.05FA0.95PbI3
Cs0.05FA0.85MA0.10Pb(I0.97Br0.03)3
Cs0.05MA0.1FA0.85PbI3
Cs0.05(FA0.95MA0.05)0.95Pb(I0.95Br0.05)3
Cs 0.05 MA 0.05 FA 0.9 Pb(I 0.95 Br 0.05 ) 3
Methylammonium Lead Iodide
FA0.95Cs0.05PbI3
4-chlorophenylethylammonium
Cs0.05(MA0.05FA0.95)0.95Pb(I0.95Br0.05)3
Cs0.05(MA0.17FA0.83)0.95Pb(I0.83Br0.17)3
MA0

perovskite_composition
- 576
- ISOD3
- formamidine formate' --> Additive

- 4-chlorophenylethylammonium --> PAPER 7, LOOK AT IT LATER
- Cl-contained formamidinium lead iodide --> 
- ammonia healed formamidinium cesium lead iodide --> NH 3 -FACsPbI 3
- CaFAMA-F --> F- Cs 0.05 FA 0.81 MA 0.14 PbI 2.55 Br 0.45

- HTAB 0.3 (FAPbI 3 ) 0.95 (MAPbBr 3 ) 0.05 --> HTAB is a large cation?
        HTAB --> n-hexyl trimethyl ammonium bromide

- formamidinium lead iodide --> FAPbI3
- Formamidinium lead triiodide --> FAPbI3
- dibutylammonium lead iodide --> (BA)2PbI 4
- methylammonium lead iodide --> MAPbI3
- FAPbI 3 with 0.03 mol fraction of both MDA and Cs cations
- Methylammonium Lead Bromide -->  MAPbBr 3
- CsMAFA -->Cs 0.05 FA 0.81 MA 0.14 PbI 2.55 Br 0.45
- Methylammonium Lead Iodide MAPbI3
- Butylammonium bromide
- formamidinium lead halides --> FAPbI3
- butylammonium lead iodide  --> BA 2 PbI 4
- Cesium lead iodide --> CsPbI 3
- CH3NH3PbI3 --> MAPbI₃
- CH 3 NH 3 PbI 3 --> MAPbI₃

In [48]:
composition = annotation[["first_num","perovskite_composition"]]
composition = composition[composition['perovskite_composition'] == "HTAB 0.3 (FAPbI 3 ) 0.95 (MAPbBr 3 ) 0.05"]
composition

first_num                     perovskite_composition
109         85  HTAB 0.3 (FAPbI 3 ) 0.95 (MAPbBr 3 ) 0.05

paper 30 - ammonia healed formamidinium cesium lead iodide NO passivating, so will be dropped and it's ok
<br>
paper 85 - HTAB 0.3 (FAPbI 3 ) 0.95 (MAPbBr 3 ) 0.05 HTAB is the passivating, fixed. 


In [ ]:
30 - 